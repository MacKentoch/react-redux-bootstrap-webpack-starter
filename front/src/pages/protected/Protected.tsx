import React from 'react';
import { RouteComponentProps } from 'react-router';
import FadeInEntrance from '../../components/fadeInEntrance';
import { OwnProps, MappedStateToProps, MappedDispatchToProps } from './index';

// #region types
export type Props = RouteComponentProps &
  MappedDispatchToProps &
  MappedStateToProps &
  OwnProps;
// #endregion

function Protected(props: Props) {
  return (
    <FadeInEntrance>
      <h1>Protected view</h1>
      <h3>If you can read, it means you are authenticated</h3>
    </FadeInEntrance>
  );
}

Protected.displayName = 'Protected';

export default Protected;
