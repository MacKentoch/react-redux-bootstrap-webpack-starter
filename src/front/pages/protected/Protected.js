// @flow

// #region imports
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};

type State = any;
// #endregion

class Protected extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <h1>Protected view</h1>
        <h3>If you can read, it means you are authenticated</h3>
      </div>
    );
  }
  // #endregion
}

export default Protected;
