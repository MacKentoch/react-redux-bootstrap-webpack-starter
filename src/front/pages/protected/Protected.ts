import React from 'react';
import FadeInEntrance from '../../components/fadeInEntrance';

// #region types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,
};
// #endregion

function Protected(props: Props) {
  return (
    <FadeInEntrance>
      <h1>Protected view</h1>
      <h3>If you can read, it means you are authenticated</h3>
    </FadeInEntrance>
  );
}

Protected.displayName = 'Protected';

export default Protected;
