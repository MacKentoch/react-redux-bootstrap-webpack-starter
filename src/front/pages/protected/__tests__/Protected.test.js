// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import Protected from '../Protected';
// #endregion

describe('Protected page', () => {
  it('renders as expected', () => {
    const props = {
      actions: {},
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <Protected {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
