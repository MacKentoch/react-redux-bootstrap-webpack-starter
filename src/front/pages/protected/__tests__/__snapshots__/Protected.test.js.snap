// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protected page renders as expected 1`] = `
<div>
  <div>
    <h1>
      Protected view
    </h1>
    <h3>
      If you can read, it means you are authenticated
    </h3>
  </div>
</div>
`;
